js/search-results.js

```diff
@@ -0,0 +1,60 @@
+class SearchResults {
+  constructor(searchListElement) {
+    this.searchListElement = searchListElement;
+  }
+
+  renderResults(companies, inputValue) {
+    for (const company of companies) {
+      this.renderSearchResultItem(company, inputValue);
+    }
+  }
+
+  renderSearchResultItem(companyData, inputValue) {
+    const {
+      companyName,
+      image: imageUrl,
+      changesPercentage,
+    } = companyData.profile;
+    const symbol = companyData.symbol;
+
+    const listElement = document.createElement("li");
+    listElement.classList.add("search-result-item");
+    const imageElement = document.createElement("img");
+    const linkElement = document.createElement("a");
+    const textElement = document.createElement("span");
+    // aa
+    // aA
+    // 'bbbb'
+    // 'Aa'
+    // 'bcc company'
+    // <mark>aa</mark>bcc company
+
+    const markedName = markValue(companyName, inputValue);
+    linkElement.innerHTML = markedName;
+
+    linkElement.setAttribute("href", `company.html?symbol=${symbol}`);
+    imageElement.setAttribute("src", `${imageUrl}`);
+    const sign = Number(changesPercentage) > 0 ? "positive" : "negative";
+
+    const markedSymbol = markValue(symbol, inputValue);
+
+    textElement.innerHTML = `(${markedSymbol}) (<span class="${sign}">${Number(
+      changesPercentage
+    ).toFixed(2)}%</span>)`;
+
+    listElement.append(imageElement, linkElement, textElement);
+    this.searchListElement.appendChild(listElement);
```

Comment:
DOM manipulation should always be done outside the loop. Here it's being executed inside of the loop (line 7)